### PR TITLE
Speed-up some tests and fixing some lint reports.

### PR DESCRIPTION
**Pull request by kozlovic**
Tests were executed against the parent commit of when they were
added to make sure that the tuning is not affecting the reproducibility
of the test. Note that some of the tests were sometimes not detecting
the bug, even when running with `-count=xxx`, so in that case, it
is difficult to know if the tuning would prevent detection if the
bug were to be reintroduced.
    
The FileStore and MsgTrace tests were also added as a matrix run 
in Travis/GA.

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

**Comment by kozlovic** [comment]
@derekcollison I started looking at some outlier in term of duration and manually tuned some of them (started with TestNoRace ones). I could not really make the overall duration fall that much. I have separated FileStore and MsgTrace tests as separate runs in Travis, not sure if this is something that you will agree with. If you do, there maybe more to do to bring some of the matrix runs down to 5/6 minutes as opposed to 9+ as they are now.

File: server/jetstream_consumer_test.go [hunk 1]
```diff
@@ -100,12 +100,21 @@ func TestJetStreamConsumerMultipleFiltersRace(t *testing.T) {
 	var seqs []uint64
 	var mu sync.Mutex
 
-	for i := 0; i < 10_000; i++ {
-		sendStreamMsg(t, nc, "one", "data")
-		sendStreamMsg(t, nc, "two", "data")
-		sendStreamMsg(t, nc, "three", "data")
-		sendStreamMsg(t, nc, "four", "data")
+	total := 10_000
+	wg := sync.WaitGroup{}
```
Comment (derekcollison) [inline server/jetstream_consumer_test.go:104]: If not using & just var wg sync.WaitGroup?

File: server/opts.go [hunk 2]
```diff
@@ -4279,16 +4279,16 @@ func parseAllowResponses(v interface{}, errors, warnings *[]error) *ResponsePerm
 }
 
 // Helper function to parse old style authorization configs.
-func parseOldPermissionStyle(v interface{}, errors, warnings *[]error) (*SubjectPermission, error) {
-	subjects, err := parsePermSubjects(v, errors, warnings)
+func parseOldPermissionStyle(v interface{}, errors *[]error) (*SubjectPermission, error) {
```
Comment (derekcollison) [inline server/opts.go:4282]: Should we move interface{} -> any in this PR too maybe?

File: server/gateway_test.go [hunk 3]
```diff
@@ -3354,8 +3354,13 @@ func getInboundGatewayConnection(s *Server, name string) *client {
 	var gwsa [4]*client
 	var gws = gwsa[:0]
 	s.getInboundGatewayConnections(&gws)
-	if len(gws) > 0 {
-		return gws[0]
+	for _, gw := range gws {
+		gw.mu.Lock()
```
[the remark this answers is not in the document]
Comment (kozlovic) [reply server/gateway_test.go:3358]: This is a `*client`, so we don't have RWMutex here.

File: server/jetstream_consumer_test.go (hunk 1 above)
Comment (kozlovic) [reply server/jetstream_consumer_test.go:104]: Sure.

File: scripts/runTestsOnTravis.sh [hunk 4]
```diff
@@ -23,6 +23,12 @@ elif [ "$1" = "no_race_tests" ]; then
 
     go test -v -p=1 -run=TestNoRace ./... -count=1 -vet=off -timeout=30m -failfast
 
+elif [ "$1" = "fs_tests" ]; then
+
+    # Run FileStore tests. By convention, all file store tests start with `TestFileStore`.
+
+    go test -race -v -run=TestFileStore ./server -count=1 -vet=off -timeout=30m -failfast
```
[the remark this answers is not in the document]
Comment (kozlovic) [reply scripts/runTestsOnTravis.sh:30]: I could, but looks like it takes only 6 seconds to run the mem store tests, so we won't save much.

File: scripts/runTestsOnTravis.sh [hunk 5]
```diff
@@ -76,13 +82,20 @@ elif [ "$1" = "mqtt_tests" ]; then
...
+    # Run Message Tracing tests. By convention, all message tracing tests start with `TestMsgTrace`.
+
+    go test -race -v -run=TestMsgTrace ./server -count=1 -vet=off -timeout=30m -failfast
+
 elif [ "$1" = "srv_pkg_non_js_tests" ]; then
 
     # Run all non JetStream tests in the server package. We exclude the
-    # JS tests by using the `skip_js_tests` build tag and MQTT tests by
-    # using the `skip_mqtt_tests`
+    # FileStgore tests by using the `skip_fs_tests` build tag, the JS tests
```
[the remark this answers is not in the document]
Comment (kozlovic) [reply scripts/runTestsOnTravis.sh:94]: Fixed

File: scripts/runTestsOnTravis.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (kozlovic) [reply scripts/runTestsOnTravis.sh:30]: I did it anyway.

File: server/opts.go (hunk 2 above)
Comment (kozlovic) [reply server/opts.go]: Hopefully this does not "break" public APIs? I know `any` is just `interface{}`, but who knows...

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by derekcollison** [comment]
Should I do normal merge or squash merge?

**Comment by kozlovic** [comment]
@derekcollison I would do simple merge (I would want the interface{}->any stay as its own commit)

**Pull request merged**